### PR TITLE
many: use interfaces.SnapAppSet in security backends

### PR DESCRIPTION
**Pull request by andrewphelpsj**
This change replaces the usage of `snap.Info` in the `SecurityBackend.Setup` and `SecurityBackendSetupMany.SetupMany` with `interfaces.SnapAppSet`.

### CODE REVIEW

File: overlord/ifacestate/handlers.go [hunk 1]
```diff
@@ -543,12 +548,18 @@ func (m *InterfaceManager) doConnect(task *state.Task, _ *tomb.Tomb) (err error)
 		return fmt.Errorf("snap %q has no %q plug", connRef.PlugRef.Snap, connRef.PlugRef.Name)
 	}
 
+	// TODO: should the repo return an app set here?
```
Comment (pedronis) [inline overlord/ifacestate/handlers.go:551]: yes, when we later do larger changes to it yes, let's make sure that TODO about this contain one consistent string like "app set" and keep note of this

**Review by pedronis — APPROVED** [review]
thank you

File: overlord/ifacestate/handlers.go [hunk 2]
```diff
@@ -711,11 +722,16 @@ func (m *InterfaceManager) doDisconnect(task *state.Task, _ *tomb.Tomb) error {
 		if err != nil {
 			return err
 		}
+
+		// i've done this a lot, i wonder if there should be/would it be
+		// possible for something like a SnapState.CurrentAppSet()?
```
Comment (bboozzoo) [inline overlord/ifacestate/handlers.go:727]: maybe convert to a TODO?

**Review by bboozzoo — APPROVED** [review]
LGTM

**Pull request merged**